SnakeGame/Snake.js

```diff
@@ -0,0 +1,207 @@
+class Snake {
+    constructor(game) {
+        this.game = game;
+        this.direction = "R";
+        this.squeue = new Queue();
+        this.head = [2, 3];
+        // start off as a 3 unit long snake
+        this.squeue.enqueue([2, 1]);
+        this.squeue.enqueue([2, 2]);
+        this.squeue.enqueue([2, 3]);
+        this.appleCount = 0;
+        this.input = "R";
```

Comment:
May want to  comment what some of these are, since their names may not be too indicative. Especially `squeue`, `direction`, and `input`.